src/github/release.go

```diff
@@ -0,0 +1,120 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/defenseunicorns/uds-releaser/src/types"
+	"github.com/defenseunicorns/uds-releaser/src/utils"
+	github "github.com/google/go-github/v66/github"
+)
+
+func TagAndRelease(flavor types.Flavor, tokenVarName string) error {
+	repo, err := utils.OpenRepo()
+	if err != nil {
+		return err
+	}
+
+	remote, err := repo.Remote("origin")
+	if err != nil {
+		return err
+	}
+
+	// Get the default branch of the current repository
+	ref, err := repo.Head()
+	if err != nil {
+		return err
+	}
```

Comment:
This functionality could be broken out since it will be the same for any git repo

Reply:
If there was an interface file this could be a static function in there